packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -0,0 +1,187 @@
+import styles from './visit-form.component.scss';
+import Button from 'carbon-components-react/es/components/Button';
+import DatePicker from 'carbon-components-react/es/components/DatePicker';
+import DatePickerInput from 'carbon-components-react/es/components/DatePickerInput';
+import Form from 'carbon-components-react/es/components/Form';
+import { useTranslation } from 'react-i18next';
+import { Column, Grid, Row } from 'carbon-components-react/es/components/Grid';
+import TimePicker from 'carbon-components-react/es/components/TimePicker';
+import TimePickerSelect from 'carbon-components-react/es/components/TimePickerSelect';
+import SelectItem from 'carbon-components-react/es/components/SelectItem';
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import Select from 'carbon-components-react/es/components/Select';
+import VisitTypeOverview from './visit-type-overview.component';
+import {
+  createErrorHandler,
+  detach,
+  getStartedVisit,
+  NewVisitPayload,
+  saveVisit,
+  showToast,
+  useLocations,
+  useSessionUser,
+  VisitMode,
+  VisitStatus,
+} from '@openmrs/esm-framework';
+import dayjs from 'dayjs';
+import { amPm, convertTime12to24 } from './start-visit-helper';
+
+interface StartVisitFormProps {
+  isTablet: boolean;
+  patientUuid: string;
+}
+
+const StartVisitForm: React.FC<StartVisitFormProps> = ({ isTablet, patientUuid }) => {
+  const { t } = useTranslation();
+  const locations = useLocations();
+  const sessionUser = useSessionUser();
+  const [selectedLocation, setSelectedLocation] = useState<string>('');
+  const [visitDate, setVisitDate] = useState<string>(dayjs(new Date()).format('DD/MM/YYYY'));
+  const [visitTime, setVisitTime] = useState<string>(dayjs(new Date()).format('hh:mm'));
+  const [timeFormat, setTimeFormat] = useState<amPm>(new Date().getHours() >= 12 ? 'PM' : 'AM');
+  const [contentSwitcherIndex, setContentSwitcherIndex] = useState<number>(1);
+  const [visitType, setVisitType] = useState<string>();
+
+  useEffect(() => {
+    if (locations && sessionUser?.sessionLocation?.uuid) {
+      setSelectedLocation(sessionUser?.sessionLocation?.uuid);
+    }
+  }, [locations, sessionUser]);
+
+  const handleCloseForm = useCallback(() => detach('patient-chart-workspace-slot', 'start-visit-workspace-form'), []);
+
+  const handleStartVisit = useCallback(
+    (event) => {
+      event.preventDefault();
+      const [hours, minutes] = convertTime12to24(visitTime, timeFormat);
+      let visitPayload: NewVisitPayload = {
+        patient: patientUuid,
+        startDatetime: new Date(
+          dayjs(visitDate).year(),
+          dayjs(visitDate).month(),
+          dayjs(visitDate).date(),
+          hours,
+          minutes,
+        ),
+        visitType: visitType,
+        location: selectedLocation,
+      };
+
+      saveVisit(visitPayload, new AbortController()).subscribe(
```

Comment:
This looks to me like a memory leak—subscribing with no `unsubscribe()`. Maybe you can use `first()`, as suggested [here](https://medium.com/javascript-everyday/no-need-to-unsubscribe-rxjs-operators-will-help-you-out-f8b8ce7bf26a)?